packages/lesswrong/components/comments/CommentsItem/BanUserFromAllPostsMenuItem.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { PureComponent } from 'react';
+import { registerComponent } from 'meteor/vulcan:core';
+import { MenuItem } from 'material-ui';
+import Users from 'meteor/vulcan:users';
+import PropTypes from 'prop-types';
+
+class BanUserFromAllPostsMenuItem extends PureComponent {
+
+  constructor(props) {
+    super(props);
+  }
+
+  handleBanUserFromAllPosts = (event) => {
+    event.preventDefault();
+    const commentUserId = this.props.comment.userId
+    let bannedUserIds = _.clone(this.props.currentUser.bannedUserIds) || []
+    if (!bannedUserIds.includes(commentUserId)) {
+      bannedUserIds.push(commentUserId)
+    }
+    this.props.userEditMutation({
+      documentId: this.props.currentUser._id,
+      set: {bannedUserIds:bannedUserIds},
+      unset: {}
+    }).then(()=>console.log(`User ${commentUserId} added to post banned-list: ${bannedUserIds}`))
+  }
+
+  render() {
+    if (this.props.comment && Users.canModeratePost(this.props.currentUser, this.props.post)) {
+      return <MenuItem className="comment-menu-item-ban-from-user" onTouchTap={ this.handleBanUserFromAllPosts } primaryText="Ban User From All Your Posts" />
```

Comment:
I think this will have to be shorter text. Even in a moderation submenu, I don't think we can fit this whole sentence into a popover menu item.

Reply:
I think we can leave it for now, and fix it with modals that make it more clear. (menu item could even just say "Ban user from commenting" and then the modal offers a choice of "on this post" and "all posts")